src/en/mangasaki/src/eu/kanade/tachiyomi/extension/en/mangasaki/MangaSaki.kt

```diff
@@ -0,0 +1,265 @@
+
+    // manga details
+    override fun mangaDetailsParse(document: Document): SManga {
+        val Manga = SManga.create()
+        Manga.author = document.select("div.field-name-field-author div.field-item").first()?.text()
+        Manga.genre = document.select("div.field-name-field-genres ul li a").joinToString { it.text() }
+        Manga.description = document.select("div.field-name-body div.field-item p").text()
+        Manga.thumbnail_url = document.select("div.field-name-field-image2 div.field-item img").attr("src")
+
+        document.select("div.field-name-field-status div.field-item").text().also { statusText ->
+            when {
+                statusText.contains("Ongoing", true) -> Manga.status = SManga.ONGOING
+                statusText.contains("Complete", true) -> Manga.status = SManga.COMPLETED
+                else -> Manga.status = SManga.UNKNOWN
+            }
+        }
+
+        return Manga
+    }
+
+    // chapters
+    override fun chapterListRequest(manga: SManga) = chapterListRequest(manga.url, 1)
+
+    private fun chapterListRequest(url: String, page: Int): Request {
+        return GET("$baseUrl$url?page=${page - 1}", headers)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        var document = response.asJsoup()
+        val chapters = document.select(chapterListSelector()).map(::chapterFromElement).toMutableList()
+        var nextPage = 2
+
+        while (document.select(paginationNextPageSelector).isNotEmpty()) {
+            val dirtyPage = document.select("div#block-search-form form#search-block-form").attr("action")
+            val cleaningIndex = dirtyPage.lastIndexOf("?")
+            val cleanPage = dirtyPage.substring(0, cleaningIndex)
+            document = client.newCall(chapterListRequest(cleanPage, nextPage)).execute().asJsoup()
+            chapters += document.select(chapterListSelector()).map(::chapterFromElement)
+            nextPage++
+        }
+
+        return chapters
+    }
+
+    private val paginationNextPageSelector = latestUpdatesNextPageSelector()
+
+    override fun chapterListSelector() = ".chlist tbody tr"
+
+    override fun chapterFromElement(element: Element): SChapter {
+        val Chapter = SChapter.create()
+        Chapter.setUrlWithoutDomain(element.select("a").attr("href"))
+        Chapter.name = element.select("a").text()
+        Chapter.date_upload = element.select("td").last()?.text()?.let {
+            SimpleDateFormat("dd MMM yyyy", Locale.ENGLISH).parse(it)?.time ?: 0
+        } ?: 0
+
+        return Chapter
+    }
+
+    // pages
+    override fun pageListParse(document: Document): List<Page> {
+        val pageList: MutableList<Page> = mutableListOf()
+        val jsonString = document.select("script:containsData(showmanga)").html().removePrefix("<!--//--><![CDATA[//><!--\njQuery.extend(Drupal.settings, ").removeSuffix(");\n//--><!]]>")
```

Comment:
```suggestion
        val jsonString = document.select("script:containsData(showmanga)").data()
            .substringAfter("(Drupal.settings, ")
            .substringBeforeLast(");")
```

Reply:
Had to change to document.select("script:containsData(showmanga)")[0].data() as the select returns a list